students/brandon_nguyen/session02/series.py

```diff
@@ -0,0 +1,90 @@
+
+#!/usr/bin/env python3
+import unittest
+#Week2 Excercise #2
+#Student: Brandon Nguyen - Au2018
+def fibonacci(n):
+    """
+    The function have one parameter n which represents the index.
+    The function returns the nth's value of the index n in fibonacci series (starting with zero indiex).
+    0, 1, 1, 2, 3, 5, 8, 13, ...
+    """
+    if (n==0 or n==1): 
+        return n
+    else:
+        return fibonacci(n-1) + fibonacci(n-2)
+def lucas(n):
+    """
+    The Lucas Numbers are a related series of integers that start with the values 2 and 1 rather than 0 and 1. 
+    The resulting series looks like this:2, 1, 3, 4, 7, 11, 18, 29, ... similiar to f(n)=f(n-1)+f(n-2)
+    The function lucas returns the nth value in the lucas numbers series (starting with zero index).
+    """
+    if (n==0 or n == 1):
+        return abs(n-2)
+    else:
+        return lucas(n-1) + lucas(n-2)
+def sum_series(n,n0=0,n1=1):   
+    """
+    compute the nth value for fibonacci or lucas serie. 
+    :param optional or default to n0=0, n1=1 determine the first two values in the series. 
+
+    if n0 == 0 and n1 == 1 do Fibbonacci 
+    if n0 == 2 and n1 == 1 do lucas
+    elif do other 
+
+    """
+    #ask the instructor which is the best way to check args.
+    if (n0==0 and n1==1):
+    #do fibonacci
+        if (n==0 or n==1): 
+            return n
+        else:
+            return fibonacci(n-1) + fibonacci(n-2)
+    elif(n0==2 and n1==1):
+    #do lucas    
+        if (n==0 or n == 1):
+            return abs(n-2)
```

Comment:
Do you see that you are returning `n0` if `n==0` and returning `n1` if `n==1`? Not just here, but also on the Fibonacci part as well (lines 39-40). This shows that you really don't need to distinguish the two series (Fibonacci and Lucas) at all. In fact, the purpose of `sum_series` is to unify all such series in one simple function, without having to check the parameter values like this and distinguished computation. Hope this helps.

Reply:
Thank you!  I have made my correction.